### PR TITLE
Unify naming scheme for macros indicating presence of classes of ciphersuites

### PR DESCRIPTION
**Pull request by hanno-arm**
__Context:__ The following helper macros indicate whether certain classes of ciphersuites are enabled:
* `MBEDTLS_SSL_SOME_SUITES_USE_CBC`
* `MBEDTLS_SSL_SOME_SUITES_USE_TLS_CBC`
* `MBEDTLS_SSL_SOME_MODES_USE_MAC`

__This PR__ unifies the naming of those macros by renaming `MBEDTLS_SSL_SOME_MODES_USE_MAC` to `MBEDTLS_SSL_SOME_SUITES_USE_MAC`.

Moreover, it introduces another macro `MBEDTLS_SSL_SOME_SUITES_USE_STREAM` controlling whether ciphersuites using stream ciphers are enabled. This check was previously inlined as `defined(MBEDTLS_ARC4_C) || defined(MBEDTLS_CIPHER_NULL_CIPHER)` throughout the code, both in `ssl_internal.h` and in the record decryption/encryption routines in `ssl_msg.c`.

Finally, all the above macros are now guarded by a check whether a pre-1.3 version of TLS is enabled, which is controlled in the newly added helper macro `MBEDTLS_SSL_PROTO_TLS1_2_OR_EARLIER`.

### CODE REVIEW

**Review by mpg — APPROVED** [review]
LGTM

File: include/mbedtls/ssl_internal.h
Comment (mpg) [inline include/mbedtls/ssl_internal.h:180]: The pre-processor complains about a missing `\` - can't entirely blame it.

**Review by mpg — CHANGES_REQUESTED** [review]
The CI found an issue I missed.

File: include/mbedtls/ssl_internal.h
Comment (hanno-arm) [reply include/mbedtls/ssl_internal.h]: Fixed.

File: library/ssl_msg.c
Comment (gabor-mezei-arm) [inline library/ssl_msg.c:762]: typo, it should be MBEDTLS_SSL_SOME_SUITES_USE_STREAM
Please correct it in all occurence below.

**Review by gabor-mezei-arm — CHANGES_REQUESTED** [review]

File: library/ssl_msg.c
Comment (hanno-arm) [reply library/ssl_msg.c]: Fixed.

**Review by gabor-mezei-arm — DISMISSED** [review]

**Comment by mpg** [comment]
Note: this conflicts with https://github.com/ARMmbed/mbedtls/pull/4156 which drops support for RC4 ciphersuites (changing the definition of ` MBEDTLS_SSL_SOME_SUITES_USE_STREAM`), and for SSLv3 (making the distinction between `MBEDTLS_SSL_SOME_SUITES_USE_CBC` and `MBEDTLS_SSL_SOME_SUITES_USE_TLS_CBC` irrelevant).

Considering 4156 is pretty big, I'm inclined to give it priority and delay this PR until 4156 has been merged. Would that be OK with you?

**Comment by hanno-arm** [comment]
@mpg Of course.

**Comment by mpg** [comment]
I'm labeling as "needs: preceding PR" then to reflect that this is blocked until #4156 is finalized and merged.

By the way, you labeled "needs: backports" - I'm not entirely sure about that, could you elaborate on that point?

**Comment by hanno-arm** [comment]
@mpg @gabor-mezei-arm I have rebased this PR on top of `development`, resolving a number of simple merge conflicts resulting from the removal of SSLv3 and weak ciphersuites. 

Could you please re-review?

**Review by gabor-mezei-arm — APPROVED** [review]

**Review by mpg — APPROVED** [review]
LGTM

**Pull request merged**
